test/s3path.jl

```diff
@@ -75,6 +75,32 @@ function test_s3_readpath(p::PathSet)
     end
 end
 
+function test_s3_walkpath(p::PathSet)
+    @testset "walkpath - S3" begin
+        # Test that we still return parent prefixes even when no "directory" objects
+        # have been created by a `mkdir`, retaining consistency with `readdir`.
+        _root = p.root / "s3_walkpath/"
```

Comment:
We should definitely have a leaf file that exists outside of this prefix. I'm pretty sure that's already the case but maybe that should be explicitly done to ensure we aren't relying too much on behaviour defined outside of this test set.

Reply:
I'm not sure I follow, we have the pathset files that we're copying into this prefix?

Reply:
We should be sure to write something outside the `_root` (e.g. `p.root / "other.txt"`) to ensure that `walkpath` doesn't doesn't traverse all prefixes in the bucket and not just those with the desired prefix.

I was also trying to mention I believe that some setup for previous tests already ensures this is the case but it's probably best to make an additional object here so that the tests in this testset are not reliant on external setup.

Reply:
> We should be sure to write something outside the _root (e.g. p.root / "other.txt") to ensure that walkpath doesn't doesn't traverse all prefixes in the bucket and not just those with the desired prefix.

We already know that's the case because we're copying the leaf files from the `p.root` prefix to the `_root` sub-prefix.